cpp/arcticdb/python/CMakeLists.txt

```diff
@@ -0,0 +1,110 @@
+if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})    
+    message(FATAL_ERROR "This file should not be included directly yet. Use the top-level CMakeLists.txt instead.")
+    find_package(arcticdb REQUIRED)
+    if(${ARCTICDB_USING_CONDA})
+    SET(ARCTICDB_CORE_LIBRARY arcticdb::arcticdb_core_shared)
+    else()
+        SET(ARCTICDB_CORE_LIBRARY arcticdb::arcticdb_core_static)
+    endif()
+else()
+    SET(ARCTICDB_CORE_LIBRARY arcticdb::arcticdb_core_static)
+endif()
+
+    
+## Core python bindings, private only ##
+set(arcticdb_python_srcs
+        ../async/python_bindings.cpp
+        ../codec/python_bindings.cpp
+        ../column_store/python_bindings.cpp
+        ../storage/python_bindings.cpp
+        ../stream/python_bindings.cpp
+        ../toolbox/python_bindings.cpp
+        ../version/python_bindings.cpp)
+
+add_library(arcticdb_python STATIC ${arcticdb_python_srcs})
+
+target_link_libraries(arcticdb_python
+        PUBLIC
+        #tsan
+        arcticdb_core_static
+        ${AWSSDK_LINK_LIBRARIES}
+        )
+
+target_include_directories(arcticdb_python PRIVATE
+        ${LIBMONGOCXX_STATIC_INCLUDE_DIRS}
+        ${LIBBSONCXX_STATIC_INCLUDE_DIRS}
+        ${BITMAGIC_INCLUDE_DIRS}
+        )
+
+
+if (NOT WIN32)
+    add_compile_options("-ftemplate-depth=1000")
+    if(NOT ${ARCTICDB_USING_CONDA})
+        # Compilers on conda-forge are relatively strict.
+        # For now, we do not want to treat warnings as errors for those builds.
+        add_compile_options("-Werror")
+    endif()
+endif ()
+
+## arcticdb_ext python module
+# This configures: linking to Python::Module + pybind, output pre-/suf-fix, -fvisibility=hidden, strip release build
+pybind11_add_module(arcticdb_ext MODULE
+       python_module.cpp
+       arctic_version.cpp
+       arctic_version.hpp
+       reader.hpp
+       adapt_read_dataframe.hpp
+        )
+
+
+set (additional_link_flags "")
```

Comment:
Do we have any convention about naming. Some variables are all caps, others are small caps. I don't find any pattern.